File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/PostApplyPowerMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import { AbstractPower } from "../Power/AbstractPower";
import AbstractCreature from "../Unit/AbstractCreature";

export class PostApplyPowerMessage implements IMessage{
    public static readonly Id = "PostApplyPowerMessage";
    constructor(public Power:AbstractPower,public Target:AbstractCreature,public Source:AbstractCreature) {
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/PostApplyPowerMessage.cs
using ModTheSpire.Power;
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    public class PostApplyPowerMessage : IEventHandle
    {
        public AbstractCreature Source;
        public AbstractCreature Target;      
        public AbstractPower Power;

        public PostApplyPowerMessage(AbstractCreature source, AbstractCreature target, AbstractPower power)
        {
            Source = source;
            Target = target;
            Power = power;
        }
    }
}